libraries/apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/internal/ResponseParser.kt

```diff
@@ -45,6 +45,10 @@ internal object ResponseParser {
 
     jsonReader.endObject()
 
+    if (jsonReader.peek() != JsonReader.Token.END_DOCUMENT) {
+      println("Apollo: extra tokens after payload")
+    }
```

Comment:
2 things:

1. Can you move this to the `toApolloResponse()` function? I'd like to keep the possibility to read a single JSON object from `ResponseParser.parse` in the future
2. Let's make it an error? For this you can `throw JsonDataException("Expected END_DOCUMENT but was ${peek()}")`

Reply:
I've made changes (hopefully) - as requested - lmk, and I'll push an equivalent to the 3.x branch.

Also - do you want the same applied to batching interceptor parsing - I'm guessing an exception should also be thrown there, not sure about whether code needs moving around?

Reply:
> I've made changes (hopefully) - as requested - lmk, and I'll push an equivalent to the 3.x branch.

Thanks, LGTM 👍 

> Also - do you want the same applied to batching interceptor parsing - I'm guessing an exception should also be thrown there, not sure about whether code needs moving around?

Yup, same exception, no need to move code around.

Reply:
ok, Ive pushed a similar change to the other 3 prs.
not sure whether I should investigate integration test failures here?

Reply:
> ok, Ive pushed a similar change to the other 3 prs.

Thanks!

> not sure whether I should investigate integration test failures here?

Doesn't look related, I'll take a deeper look tomorrow

Reply:
Turns out it was related. `peek()` can throw but `toApolloResponse()` shouldn't. See https://github.com/apollographql/apollo-kotlin/pull/5895/commits/9ab202cc3536b505a8387f0e619db3660cf2ba62

I'll take it from there for this PR and the other 3. Thanks again for looking into this!